apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/page.tsx

```diff
@@ -48,65 +54,87 @@ export default async function DocumentPage({
   const { workspace } = await getCurrentUser()
   const { projectId: pjid, commitUuid, documentUuid } = await params
   const projectId = Number(pjid)
+
+  const documentLogsRepo = new DocumentLogsRepository(workspace.id)
+  if (!documentLogsRepo.hasLogs(documentUuid)) {
+    return <DocumentLogBlankSlate />
+  }
+
   const commit = await findCommitCached({ projectId, uuid: commitUuid })
-  const { logUuid, pageSize, page: pg } = await searchParams
+  const commits = await findCommitsByProjectCached({ projectId })
+
+  const {
+    logUuid,
+    pageSize,
+    page: pageString,
+    versions: _selectedCommitsIds,
+    origins: _selectedLogSources,
+  } = await searchParams
+
+  const originalSelectedCommitsIds = [
+    ...commits.filter((c) => !!c.mergedAt).map((c) => c.id),
+    ...(!commit.mergedAt ? [commit.id] : []),
+  ]
+
+  const selectedCommitsIds = (
+    Array.isArray(_selectedCommitsIds)
+      ? _selectedCommitsIds
+      : (_selectedCommitsIds?.split(',') ?? originalSelectedCommitsIds)
+  ).map(Number)
+
+  const selectedSources = (
+    Array.isArray(_selectedLogSources)
+      ? _selectedLogSources
+      : (_selectedLogSources?.split(',') ?? Object.values(LogSources))
+  ).filter((s) =>
+    Object.values(LogSources).includes(s as LogSources),
+  ) as LogSources[]
+
+  const logsFilterOptions: DocumentLogFilterOptions = {
+    commitIds: selectedCommitsIds,
+    logSources: selectedSources,
+  }
+
   const documentLogUuid = logUuid?.toString()
-  const page = pg?.toString?.()
+  const page = pageString?.toString?.()
   const currentLogPage = await fetchDocumentLogPage({
     workspace,
-    commit,
+    filterOptions: logsFilterOptions,
     documentLogUuid,
   })
 
   if (currentLogPage && currentLogPage !== page) {
     const route = ROUTES.projects
       .detail({ id: projectId })
       .commits.detail({ uuid: commit.uuid })
-      .documents.detail({ uuid: documentUuid }).logs.root
-    return redirect(
-      `${route}?page=${currentLogPage}&logUuid=${documentLogUuid}`,
-    )
+      .documents.detail({ uuid: documentUuid }).logs
+
+    const parameters = [
```

Comment:
you created a "addParameter" function why not use it here? Although you could use URLSearchParams

Reply:
Name of parameters are different between api and web. I would keep this as it is.